src/Atmos/Parameterizations/CloudPhysics/Microphysics.jl

```diff
@@ -1,271 +1,575 @@
+const ACloudPS  = AbstractCloudParameterSet
+const APrecipPS = AbstractPrecipParameterSet
+const ALPS = AbstractLiquidParameterSet
+const AIPS = AbstractIceParameterSet
+const ARPS = AbstractRainParameterSet
+const ASPS = AbstractSnowParameterSet
+
+export ζ_rai
+export n0_sno
+export τ_relax
+export lambda
+export unpack_parameters
+
+export supersaturation
+export G_func
 
-# rain fall speed
-export terminal_velocity_single_drop_coeff
 export terminal_velocity
 
-# rates of conversion between microphysics categories
-export conv_q_vap_to_q_liq
-export conv_q_vap_to_q_ice
-export conv_q_liq_to_q_rai_acnv
-export conv_q_liq_to_q_rai_accr
-export conv_q_rai_to_q_vap
+export conv_q_vap_to_q_liq_ice
+
+export conv_q_liq_to_q_rai
+export conv_q_ice_to_q_sno
+
+export accretion
+export accretion_rain_sink
+export accretion_snow_rain
+
+export evaporation_sublimation
+export snow_melt
 
 """
-    terminal_velocity_single_drop_coeff(ρ)
+    ζ_rai(param_set, ρ)
 
-where:
-  - `ρ` is the density of air
+ - `param_set` - abstract set with earth parameters
+ - `ρ` air density
 
 Returns the proportionality coefficient between terminal velocity of an
-individual water drop and the square root of its radius * g.
+individual water drop and the square root of its radius.
 """
-function terminal_velocity_single_drop_coeff(
-    param_set::APS,
-    ρ::FT,
-) where {FT <: Real}
+function ζ_rai(param_set::APS, ρ::FT) where {FT<:Real}
+
     _ρ_cloud_liq::FT = ρ_cloud_liq(param_set)
-    _C_drag::FT = C_drag(param_set)
+    _C_drag::FT      = Microphysics.C_drag(param_set)
+    _grav::FT        = grav(param_set)
+
+    #@info(ρ)
+
+    return sqrt(_grav * FT(8/3) / _C_drag * (_ρ_cloud_liq / ρ - FT(1)))
+end
+
+"""
+    n0_sno(snow_param_set, q_sno, ρ)
+
+ - `snow_param_set` - abstract set with snow parameters
+ - `q_sno` -  snow specific humidity
+ - `ρ` - air density
 
-    # terminal_vel_of_individual_drop = v_drop_coeff * (g * drop_radius)^(1/2)
-    return sqrt(FT(8 / 3) / _C_drag * (_ρ_cloud_liq / ρ - FT(1)))
+Returns the intercept parameter of the assumed Marshall Palmer distribution of
+snow particles.
+"""
+function n0_sno(snow_param_set::ASPS, q_sno::FT, ρ::FT) where {FT<:Real}
+
+    _ν_sno::FT = ν_sno(snow_param_set)
+    _μ_sno::FT = μ_sno(snow_param_set)
+
+    return _μ_sno * (ρ * q_sno)^_ν_sno
 end
 
 """
-    terminal_velocity(q_rai, ρ)
+    τ_relax(cloud_param_set)
+
+ - `cloud_param_set` - abstract set with cloud liquid water or cloud ice
+    microphysics parameters
+
+Returnts the relaxation timescale for condensation and evaporation of
+cloud liquid water or the relaxation timescale for sublimation and
+resublimation of cloud ice.
+"""
+function τ_relax(liquid_param_set::ALPS)
+
+    _τ_relax = τ_cond_evap(liquid_param_set)
+    return _τ_relax
+end
+function τ_relax(ice_param_set::AIPS)
 
-where:
-  - `q_rai` - rain water specific humidity
-  - `ρ`     - density of air
+    _τ_relax = τ_sub_resub(ice_param_set)
```

Comment:
```suggestion
    _τ_relax = τ_subl_depo(ice_param_set)
```
(and consistently throughout?)

Reply:
Would prefer to make this "depo_subl" (or "dep_sub") to match "cond_evap"
```suggestion
    _τ_relax = τ_dep_sub(ice_param_set)
```

Reply:
done